pkg/sql/schemachanger/scplan/internal/scgraph/dep_edge_tree.go

```diff
@@ -11,100 +11,217 @@
 package scgraph
 
 import (
+	"sync"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/screl"
 	"github.com/cockroachdb/cockroach/pkg/util/iterutil"
+	"github.com/cockroachdb/errors"
 	"github.com/google/btree"
 )
 
-type depEdgeTree struct {
-	t     *btree.BTree
-	order edgeTreeOrder
-	cmp   nodeCmpFn
-}
-
-type nodeCmpFn func(a, b *screl.Node) (less, eq bool)
-
-func newDepEdgeTree(order edgeTreeOrder, cmp nodeCmpFn) *depEdgeTree {
-	const degree = 8 // arbitrary
-	return &depEdgeTree{
-		t:     btree.New(degree),
-		order: order,
-		cmp:   cmp,
-	}
-}
-
 // edgeTreeOrder order in which the edge tree is sorted,
 // either based on from/to node indexes.
 type edgeTreeOrder bool
 
-func (o edgeTreeOrder) first(e Edge) *screl.Node {
-	if o == fromTo {
-		return e.From()
-	}
-	return e.To()
-}
-
-func (o edgeTreeOrder) second(e Edge) *screl.Node {
-	if o == toFrom {
-		return e.From()
-	}
-	return e.To()
-}
-
 const (
 	fromTo edgeTreeOrder = true
 	toFrom edgeTreeOrder = false
 )
 
-// edgeTreeEntry BTree items for tracking edges
-// in an ordered manner.
-type edgeTreeEntry struct {
-	t    *depEdgeTree
-	edge *DepEdge
+// getTargetIdxFunc is used to get the location of the node in the graph.
+type getTargetIdxFunc = func(n *screl.Node) targetIdx
+
+// depEdges is a data structure to store the set of depEdges. It offers
+// in-order traversal of edges from or to any node in the graph.
+type depEdges struct {
+	fromTo *btree.BTree
+	toFrom *btree.BTree
```

Comment:
Do you think `depEdgesByFrom` (or `depEdgesFrom`, or `edgesFrom`, or just `from`) is a better name?

Reply:
I sort of liked how it was parallel to the order of the entries in the trees. I don't feel strongly. Maybe others can opine.